app/src/main/java/com/keylesspalace/tusky/StatusListActivity.kt

```diff
@@ -221,17 +229,23 @@ class StatusListActivity : BottomSheetActivity(), HasAndroidInjector {
         val tag = hashtag ?: return true
 
         lifecycleScope.launch {
+            var filterCreateSuccess = false
+            val hashedTag = if (tag.startsWith('#')) tag else "#" + tag
+
             mastodonApi.createFilter(
                 title = "#$tag",
                 context = listOf(FilterV1.HOME),
                 filterAction = Filter.Action.WARN.action,
                 expiresInSeconds = null
             ).fold(
                 { filter ->
-                    if (mastodonApi.addFilterKeyword(filterId = filter.id, keyword = tag, wholeWord = true).isSuccess) {
-                        mutedFilter = filter
-                        updateTagMuteState(true)
+                    if (mastodonApi.addFilterKeyword(filterId = filter.id, keyword = hashedTag, wholeWord = true).isSuccess) {
+                        // must be requested again; otherwise does not contain the keyword (but server does)
+                        mutedFilter = mastodonApi.getFilter(filter.id).getOrNull()
+
+                        // TODO the preference key here ("home") is not meaningful; should probably be another event if any
                         eventHub.dispatch(PreferenceChangedEvent(filter.context[0]))
+                        filterCreateSuccess = true
                     } else {
                         Snackbar.make(binding.root, getString(R.string.error_muting_hashtag_format, tag), Snackbar.LENGTH_SHORT).show()
```

Comment:
I'd remove the '#' from the format string and just pass in `hashedTag` here. It's a little confusing that the code uses `hashedTag` everywhere, then reverts to `tag` here.

Also, per the earlier check, `tag` may already start with a `#`, so including the `#` in the format string means it would appear doubled.

Reply:
(tried to use `tag` for everything/all outputs and `hashedTag` only for the actual api code and comparisson.)

Reply:
It would probably be wise to just remove the "if" in the declaration of `hashedTag`. The current code always assumes that `tag` (this.hashtag) is without #.

Otherwise I would also change the error code strings (which would be a mess).

Reply:
Ok, did that now (remove if)